### ISSUE TITLE
getting NaN loss

### ISSUE THREAD

**Issue by ankanbhunia** [lucidrains/denoising-diffusion-pytorch#61]
I tried two datasets that consists of 10k and 30k images respectively. Can you please tell me if I need to change any hyperparameter?

**Comment by inooni** [comment]
@trufty It works for me. Thank you !!

**Comment by jwuphysics** [comment]
Unfortunately I'm still getting this error (with `amp=True`) on v0.27.4.

**Comment by KANGXI123456** [comment]
Which dataset are you running on? I have similar issues on CelebA_align, and learn_lr=1e-5. I am getting inf loss. on v0.27.4

> [12f95b3](https://github.com/lucidrains/denoising-diffusion-pytorch/commit/12f95b33d8a2f44ffb48ec1079083ec634c4728f) seems to have fixed this issue for me. I can set `amp = True` again without NaN.

**Comment by jwuphysics** [comment]
I'm using a custom astronomical dataset. I may have set the lr too high (was at 1e-4), but in any event this worked with `amp=False`.

Also, I tried to load the state dict for the model + optimizer before everything went NaN, but I wasn't able to train again -- the loss was still stuck at NaN

**Comment by p-sodmann** [comment]
I am getting NaN Losses with the learned positional embeddings, when turning it off, the loss is fine

**Comment by heitorrapela** [comment]
I was getting NAN for a custom dataset under the amp = True flag also (version [0.27.9](https://github.com/lucidrains/denoising-diffusion-pytorch/commit/e0a1bed31a29cfc1170c83bd99d5d5b2edefcabf)). So, I changed the flag to false and the training is doing good.
